### PR TITLE
Add enchanced faultInjection config

### PR DESCRIPTION
**Pull request by alexshtin**
## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: templates/server-configmap.yaml [hunk 1]
```diff
@@ -39,11 +39,17 @@ data:
             {{- toYaml . | nindent 12 }}
             {{- end }}
           {{- end }}
-          {{- if $.Values.server.config.persistence.faultinjection}}
-          {{- if $.Values.server.config.persistence.faultinjection.rate }}
+          {{- if or $.Values.server.config.persistence.default.faultInjection $.Values.server.config.persistence.faultInjection}}
```
Comment (alexshtin) [inline templates/server-configmap.yaml:42]: Old simple "just `rate`" format is still supported to keep backward compatibility.

File: values/values.cassandra.yaml [hunk 2]
```diff
@@ -8,6 +8,24 @@ server:
       default:
         driver: "cassandra"
 
+#        faultInjection:
```
Comment (alexshtin) [inline values/values.cassandra.yaml:11]: I leave it here for future references as example on how to configure fault injection.

File: templates/server-configmap.yaml (hunk 1 above)
Comment (mindaugasrukas) [reply templates/server-configmap.yaml:42]: We may not need this backward compatibility.
I'd go with a simple version without backward compatibility.

File: templates/server-configmap.yaml [hunk 3]
```diff
@@ -39,11 +39,17 @@ data:
...
           faultInjection:
-            rate: {{ $.Values.server.config.persistence.faultinjection.rate }}
-          {{- end }}
+            {{- if $.Values.server.config.persistence.faultInjection}}
+            {{- if $.Values.server.config.persistence.faultInjection.rate }}
+            rate: {{ $.Values.server.config.persistence.faultInjection.rate }}
+            {{- end }}
+            {{- else if $.Values.server.config.persistence.default.faultInjection}}
```
Comment (mindaugasrukas) [inline templates/server-configmap.yaml:48]: `if` is not necessary for `with`: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

**Review by mindaugasrukas — APPROVED** [review]

**Pull request merged**